src/vma/sock/sockinfo_udp.cpp

```diff
@@ -618,6 +623,8 @@ int sockinfo_udp::connect(const struct sockaddr *__to, socklen_t __tolen)
 			m_p_connected_dst_entry->set_so_bindtodevice_addr(m_so_bindtodevice_ip);
 		}
 
+		m_was_connected = true; // will inspect for SRC
```

Comment:
This parameter needs to be set _before_ the call to `on_sockname_change()` since it's used before returning from this function to determine the value of `m_rx_packet_processor`.
Please move this to before `on_sockname_change()`.